### PR TITLE
Cleaner Buttons

### PR DESCRIPTION
**Pull request by SirajChokshi**
## Status:

:rocket: Ready

## Description

- Button Changes
  - Reduced padding (inferred from size)
  - Added button sizes (sm/md/lg) with scaled border radius/width
  - Swapped unicode arrow for icon
- Moved font stacks to CSS variables

## Screenshots

| New | Old
| - | - |
| Screenshot 2022-10-20 at 15 14 45|Screenshot 2022-10-20 at 15 15 40|
|Screenshot 2022-10-20 at 15 16 53|Screenshot 2022-10-20 at 15 18 03|

### CODE REVIEW

**Comment by SirajChokshi** [comment]
@AndrewLester for some reason prefetched anchor tags are not styled with `text-decoration: none;` (example: https://uiuc-website-git-siraj-buttons-hack4impact1.vercel.app/join/sponsors)

Any idea why this might be happening?

Screenshot 2022-10-20 at 15 23 43@2x

**Comment by AndrewLester** [comment]
I think it's happening because the `text-decoration: none` property is on the child of the `a` tag, and after some quick testing `text-decoration: none` on a child won't override a `text-decoration: underline` on a parent (if both have a decoration, the decorations STACK). 

I'm kinda thinking we might want to either:
- Let the user choose which element they want the `Button` component to render as (`a` or `button`), and pass in attribute accordingly. This way, we can style the `a` directly in `Button.svelte`.
- Lose the button component and just use HTML elements with global classnames/styles (to style`a` and `button` the same). The downside to this approach is we lose code splitting for the button CSS and the variant autocomplete is now just CSS classnames to memorize.

Both approaches avoid `a` and `button` nesting, which I think is itself an issue: [anchor spec](https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-a-element) (No interactive elements as children, which includes `button`.)

Thoughts? Also, for an example of a prefetch link with no decoration, check the learn more buttons [here](https://uiuc-website-git-siraj-buttons-hack4impact1.vercel.app/projects).

**Comment by SirajChokshi** [comment]
> Let the user choose which element they want the `Button` component to render as (`a` or `button`), and pass in attribute accordingly. This way, we can style the `a` directly in `Button.svelte`.

I'm a fan of this approach, [similar to what IBM Carbon does](https://carbon-components-svelte.onrender.com/components/Button#link-button). Funnily enough, theres only one place the `<Button />` component is used for anything other than wrapping it with a link ([for showing more alum](https://github.com/hack4impact-uiuc/website/blob/9a7cd033fbfad65f792ce298a179cd264b432ecd/src/routes/about/team.svelte#L88)).

I'll move forward with adding conditional rendering for Link Buttons.

File: src/components/Button.svelte [hunk 1]
```diff
@@ -7,81 +12,129 @@
...
+    href?.startsWith("http") ||
+    href?.startsWith("mailto") ||
+    href?.startsWith("tel");
+</script>
+
+<svelte:element
+  this={tag}
+  {...props}
+  on:click
+  sveltekit:prefetch={!isExternal ? true : undefined}
```
Comment (AndrewLester) [inline src/components/Button.svelte]: Really wanna update SvelteKit for this: https://kit.svelte.dev/docs/link-options#data-sveltekit-prefetch

I might go for it soon, SvelteKit is now in release candidate phase so they won't push anything causing an entire repo refactor again.

**Review by AndrewLester — APPROVED** [review]

**Pull request merged**